main.tf

```diff
@@ -194,8 +212,9 @@ resource "time_sleep" "wait_for_authorization_policy" {
 ##############################################################################
 
 resource "ibm_is_vpc_routing_table" "route_table" {
-  for_each                      = module.dynamic_values.routing_table_map
-  name                          = var.prefix != null ? "${var.prefix}-${var.name}-route-${each.value.name}" : "${var.name}-route-${each.value.name}"
+  for_each = module.dynamic_values.routing_table_map
+  # Use var.routing_table_name if not null, otherwise, use var.prefix or var.name.
+  name                          = var.routing_table_name != null ? "${var.routing_table_name}-${each.value.name}" : (var.prefix != null ? "${var.prefix}-route-${each.value.name}" : "${var.name}-route-${each.value.name}")
```

Comment:
If the consumer passed a value for `prefix` and for `name` (and no value for `routing_table_name`), then I think we will want the routing table name name to be called `${var.prefix}-${var.name}-route-${each.value.name}`. Currently logic does not take that into consideration.